Update managing accounts guide (#1778)
LiskHQ/lisk-docs: docs/modules/ROOT/pages/run-blockchain/account-management.adoc

// Settings
:idprefix:
:idseparator: -
:toc: preamble
//URLs
:url_wiki_Ed25519: https://en.wikipedia.org/wiki/EdDSA#Ed25519
:url_eip2334: https://eips.ethereum.org/EIPS/eip-2334
:url_eip2333: https://eips.ethereum.org/EIPS/eip-2333
:url_rfc8032: https://datatracker.ietf.org/doc/html/rfc8032#section-5.1.5
:url_lisk_wallet: https://lisk.com/wallet
:url_github_bip_39: https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#generating-the-mnemonic
:url_github_bip_173: https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki
:url_github_lip_18: https://github.com/LiskHQ/lips/blob/master/proposals/lip-0018.md#step-3-base32
:url_github_lip_66_getprivatekeyfromphraseandpath: https://github.com/LiskHQ/lips/blob/main/proposals/lip-0066.md#getprivatekeyfromphraseandpath
// Project URLs
:url_core_cli: v4@lisk-core::core-cli.adoc
:url_sdk_cli: v6@lisk-sdk::client-cli.adoc
:url_sdk_keyscreate: {url_sdk_cli}#keyscreate
//Footnotes
:fn_bip173: footnote:bip173[See {url_github_bip_173}[BIP173^]]

====
On this page, you'll learn how to  manage your account directly from the command-line with the CLI of a Lisk node.

> Specifically, you'll learn about:

Author: ```suggestion
Specifically, the following topics below will be covered in detail:
```

|lsk7dqp2w7447pqhuzmhx
sxr65qeoq7nbfox28wxc
a|The address is the main public identifier of the account and the human-readable representation of the `publicKey`.

Depending on the context, the address is represented in <<account-address-formats,different formats>>.

|`publicKey`
|0ca6a5d44cc366de186a2589e3e51849272fa0ae9c52646766ddb7820dfe770c
|A {url_wiki_Ed25519}[Ed25519^] public key.
The public identifier of the account.
Derived from the `passphrase`, together with the `keyPath`.
The first part of the public key is used to derive the `address` for an account.

|`privateKey`
|bd7e42529df375cb9133ddf299641d38975fa4d0930eff6e7c072d4d3ff190830ca6a5d44cc366de186a2589e3e51849272fa0ae9c52646766ddb7820dfe770c
|{url_wiki_Ed25519}[Ed25519^] private key
The private identifier of the account.
Derived from the `passphrase`, together with the `keyPath`.
|===

.Credentials required for block generation
[cols="1,1,3",options="header",stripes="hover"]
|===
|`generatorKeyPath`
|"m/25519'/134'/0'/0'"
|The derivation path for the <<generator-keys>>.

|`generatorKey`
|e59f2f603b84b9cd91e2
67240a342ba09adaad348e
b984cc6e21c14fd02f28e7
a|The public <<generator-keys,generator key>>.
[#generator-keys]

> An extra {url_wiki_Ed25519}[Ed25519^] key pair, the generator keys, is added to a validator account to sign blocks.

Author: Imho it is correct to use "is" here, because it refers to a pair of keys.

A validator node only requires access to the private generator key, but not to the private key that is used for signing transactions.

> This mitigates the risk, that if an attacker is able to get the encrypted passphrase and the password used for encryption, the attacker has full control over the validator account.

Author: ```suggestion
This mitigates the risk, whereby an attacker could acquire the encrypted passphrase and the password used for encryption, and gain full control over the validator account.
```

--
Lisk app::
+
--
[source,bash]
----
./bin/run keys:create --passphrase your-passphrase --no-encrypt --count 2 --offset 1
----
--
=====

The same passphrase is used for every account, the only thing that will change is the <<key-derivation-path>>, for example when signing transactions.

[NOTE]
====
If a user uses the *default key-derivation path* `m/44'/134'/0`, it is not necessary to mention it as a parameter when the account credentials are requested.

If you are using a custom key-derivation path, make sure to always specify the key derivation path when providing the passphrase.
For example, for the `transaction:create` command, the key derivation path can be specified like so:

 % ./bin/run transaction:create hello createHello 10000000 --key-derivation-path="m/44'/134'/1'" --params='{"message":"Hello world!"}'

====

== Account initialization

To create a new account in the blockchain network, it is necessary to initialize the account.

A new account is initialized by sending tokens to an address for the first time.

When tokens are sent to a new address, it is required to pay an *account initialization fee*.

In the Lisk Mainnet, the account initialization fee is 0.05 LSK.

> The account initialization fee has to be paid on top of the other fees for a transaction.

Author: ```suggestion
NOTE: The account initialization fee has to be paid on top of the other fees for a transaction, when tokens are sent to a new address for the first time.
```

After the address received first tokens, the address is stored in the user substore.

=== User Substore

> The user substore is a data store of the Token module and contains entries dedicated to storing the balances of users for a given address and tokenID.

Author: ```suggestion
The Token module features a user substore that comprises entries exclusively intended for storing user balances for a specific address and tokenID.
```

==== Lisk32 address computation
The procedure to obtain the Lisk32 representation of an <<hexadecimal-representation>> of an address is as follows:

. The <<hexadecimal-representation,hex address>> is used to create a 30-bit checksum of the address, using a BCH code{fn_bip173}.
This step provides protection against accidental typing mistakes.
. Concatenate the address and the output of step 1.
. Encode the output of step 2 in a custom Base32 format, see {url_github_lip_18}[LIP 0018^] for details.
Lower-case letters and digits are used, the characters i, l, 1 and 0 are removed for usability reasons.
. Add the prefix "lsk" to the output of step 3.

NOTE: The Lisk32 representation always tarts with "lsk" and the final address is 41 characters long.

== Key Management
There are two recommended ways to manage the account keys, depending on the use case:

.Account credentials
[cols="1,2",options="header",stripes="hover"]
|===
|Key management method
|Use Case

|Using a secret passphrase
|For users that need their key pair only on local machines, e.g., for singing transactions.

> Requires to remember or store secretly a secret recovery phrase and a password, but does not require to store and backup any encrypted data locally.

Author: ```suggestion
It is necessary to securely store or remember a secret recovery phrase and password. However, it is not necessary to back up and store any encrypted data locally.
```
